pkg/prometheus/resource_selector.go

```diff
@@ -1328,6 +1328,19 @@ func (rs *ResourceSelector) validateLinodeSDConfigs(ctx context.Context, sc *mon
 }
 func (rs *ResourceSelector) validateKumaSDConfigs(ctx context.Context, sc *monitoringv1alpha1.ScrapeConfig) error {
 	for i, config := range sc.Spec.KumaSDConfigs {
+		if config.ClientID != nil && rs.version.LT(semver.MustParse("2.50.0")) {
+			return fmt.Errorf("field `config.ClientID` is only supported for Prometheus version >= 2.50.0")
+		}
+
+		parsedURL, err := url.Parse(config.Server)
+		if err != nil {
+			return err
+		}
+
+		if len(parsedURL.Scheme) == 0 || len(parsedURL.Host) == 0 {
```

Comment:
IIUC Prometheus only supports http and https schemes so we can check for these values. It can also be added to the API validation rules (e.g. config.Server should start by http:// or https://).

Reply:
> Prometheus doesn't expect a particular kind of scheme for the server URL. See here: https://github.com/prometheus/prometheus/blob/main/discovery/xds/kuma.go#L84-L86

I suggested this change since I wasn't sure if other schemes are also accepted.